File: diff-adapter/src/main/java/ru/starksoft/differ/adapter/ViewHolderFactory.java
package ru.starksoft.differ.adapter;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import ru.starksoft.differ.adapter.viewholder.DifferViewHolder;

@FunctionalInterface
public interface ViewHolderFactory {

    @NonNull
    DifferViewHolder create(ViewGroup parent, int viewType, OnClickListener onClickListener);
}

File: diff-adapter/src/main/java/ru/starksoft/differ/adapter/ViewHolderFactory.kt
package ru.starksoft.differ.adapter

import android.view.ViewGroup
import ru.starksoft.differ.adapter.viewholder.DifferViewHolder
import ru.starksoft.differ.adapter.viewmodel.DifferViewModel

fun interface ViewHolderFactory {

    fun create(
        parent: ViewGroup,
        viewType: Int,
        onClickListener: OnClickListener?
    ): DifferViewHolder<out DifferViewModel>
}